rules/repositories.bzl

```diff
@@ -59,14 +69,29 @@ def rules_ios_dependencies():
         sha256 = "200a35c2c84096f155a5c6092bac2abf1b2149fbeb3eb730a2923997db079a83",
     )
 
-    _maybe(
-        github_repo,
-        name = "build_bazel_rules_apple",
-        ref = "b43c7f60b584d68d7d187c236d4328162ba2e806",
-        project = "bazelbuild",
-        repo = "rules_apple",
-        sha256 = "d3e8549c0c8966ba0e547a02f5601440be25a0b8143bd57ab5834b65b02c22be",
-    )
+    bazel_version = _get_bazel_version()
+    if bazel_version.major == "5":
```

Comment:
I think it would be helpful for others to add instructions in the `README` or as comments here on how to proceed to bump `rules_apple` in the future. It might be obvious what to do for some but IMO we shouldn't assume that that's the case for all contributors.

Reply:
My current thinking about this is in the README ATM we can expand it
> the burden to be an outlier should fall on the outliers here.

In practice this is less of an issue now that the PR allows us to run many versions of Bazel with rules_ios HEAD and from a maintainer perspective caring about 6.0 as long as the `rules_apple_api` is upheld. If the API `rules_apple_api_version` is divergent, then you'll just need to add a simple if statement for the new API. You can think about this in a similar way as large scale migrations or similar to code that supports on 2 python versions. 

e.g. @mattrobmattrob  they can continue to bump rules_apple and generally not have to reason about how to back porting features for LTS support and our CI vets that it's green. 

I estimate burning this LTS 5.0 support pretty anyhow - say 6 months. We kept around 4.0 support for a similar time